src/theory/strings/term_registry.cpp

```diff
@@ -203,8 +203,11 @@ void TermRegistry::preRegisterTerm(TNode n)
   }
   else if (tn.isBoolean())
   {
-    // Get triggered for both equal and dis-equal
-    ee->addTriggerPredicate(n);
+    if (k==STRING_STRCTN || k == STRING_LEQ)
```

Comment:
Perhaps you could document in a comment why you are restricting only to these two predicates and excluding the rest of the predicates?